client/tasks/fills/PaymentGatewaySuggestions/index.js

```diff
@@ -121,7 +122,19 @@ export const PaymentGatewaySuggestions = ( { onComplete, query } ) => {
 		updatePaymentGateway( id, {
 			enabled: true,
 		} ).then( () => {
-			onComplete();
+			onComplete(
+				// use the paymentGateways variable.
+				// gateway variable doesn't have hasPlugins property.
+				paymentGateways.get( id )?.hasPlugins === undefined
```

Comment:
https://github.com/woocommerce/woocommerce-admin/blob/9600d87b2497c4a1b901b363c686b9cb8c2fdf16/client/tasks/fills/PaymentGatewaySuggestions/index.js#L73

@moon0326: How about adding `!!`  to ensure `hasPlugins` is a boolean? Then we don't need to compare with `undefined` here.

Reply:
That's a good idea 👍 Updated in https://github.com/woocommerce/woocommerce-admin/pull/8389/commits/4a3576de5d45aab3cb123c86c4e284e456e646d2 and https://github.com/woocommerce/woocommerce-admin/pull/8389/commits/4f9172c3496b5b270981e665f81379ce58cee008

Reply:
Oops, @moon0326 Sorry, I think we should invert the condition operator. Maybe also update the comment. 🙂 


```js
paymentGateways.get( id )?.hasPlugins
  ? {}
  :  {
      redirectPath: getNewPath(
        { task: 'payments' },
        {},
        '/'
      ),
    }
```

or

```js
! paymentGateways.get( id )?.hasPlugins
  ? {
      redirectPath: getNewPath(
        { task: 'payments' },
        {},
        '/'
      ),
    }
  : {}
```

Reply:
🤦 Sorry about that. I should have re-tested it after the change. Updated in https://github.com/woocommerce/woocommerce-admin/pull/8389/commits/c390e0af407e536fcc06f8bb088b5a3017ff4583